File: server/model/post.js
import { DataTypes, Model } from "sequelize";

export default (sequelize) => {
  class Post extends Model {}

  Post.init(
    {
      title: {
        type: DataTypes.STRING,
        allowNull: false,
        validate: {
          notEmpty: { msg: "Post title is required" },
        },
      },
      content: {
        type: DataTypes.TEXT,
        allowNull: false,
        validate: {
          notEmpty: { msg: "Post content is required" },
        },
      },
      categories: {
        type: DataTypes.STRING,
        allowNull: true,
      },
      isApproved: {
        type: DataTypes.BOOLEAN,
        defaultValue: false,
      },
      approverId: {
        type: DataTypes.INTEGER, // or DataTypes.UUID if using UUIDs
        allowNull: true, // Can be null if not yet approved
        references: {
          model: "users", // Reference the 'users' table
          key: "id",
        },
      },
      status: {
        type: DataTypes.ENUM("draft", "published", "archived"),
        defaultValue: "draft",
      },
      views: {
        type: DataTypes.INTEGER,
        defaultValue: 0,
      },
      featuredImage: {
        type: DataTypes.STRING,
      },
      authorId: {
        type: DataTypes.INTEGER, // or DataTypes.UUID
        allowNull: false,
        references: {
          model: "users",
          key: "id",
        },
      },
    },
    {
      sequelize,
      paranoid: true,
      modelName: "Post",
      tableName: "posts",
      timestamps: true,
      deletedAt: "destroyTime",
    }
  );

  return Post;
};

File: server/Models/Post.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace blogapp.Models
{
    public class Post
    {
        [Key]
        public int Id { get; set; }

        [Required(ErrorMessage = "Post title is required")]
        [StringLength(255)]
        public string Title { get; set; }

        [Required(ErrorMessage = "Post content is required")]
        [Column(TypeName = "TEXT")]
        public string Content { get; set; }

        [StringLength(255)]
        public string? Categories { get; set; }

        [Column("isApproved")]
        public bool IsApproved { get; set; } = false;

        [ForeignKey("Approver")]
        public int? ApproverId { get; set; }
        public virtual User? Approver { get; set; }

        [Column(TypeName = "nvarchar(50)")]
        public string Status { get; set; } = "draft";

        public int? Views { get; set; } = 0;

        [StringLength(255)]
        public string? FeaturedImage { get; set; }

        [Required]
        [ForeignKey("Author")]
        public int AuthorId { get; set; }
        public virtual User Author { get; set; }

        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        [Column("destroyTime")]
        public DateTime? DeletedAt { get; set; }
    }
}